soroban-env-host/src/builtin_contracts/stellar_asset_contract/contract.rs

```diff
@@ -51,6 +51,30 @@ fn check_non_native(e: &Host) -> Result<(), HostError> {
     }
 }
 
+fn check_not_issuer(e: &Host, addr: &Address) -> Result<(), HostError> {
```

Comment:
FWIW a safer/more idiomatic way to implement this would be to get issuer address, then compare it to the `addr` via `host.compare`. This way you wouldn't miss metering on `==` operation (which has an insignificant cost of course, but still some cost). I also suspect that's not the only place where we could use the issuer address. But probably we don't have time for this, IDK really. Maybe we need to make a quick pass to update this everywhere...

Reply:
Ah good point. I've updated to use compare in this PR, but I'm not sure if it's worth going through and looking/fixing the other uses.